pkg/image/image.go

```diff
@@ -59,3 +72,137 @@ func ParseReference(ref string) (imagesource.TypedImageReference, error) {
 	}
 	return imagesource.TypedImageReference{Ref: dst, Type: dstType}, nil
 }
+
+/* GetConfigDirFromOCI verifies the ref is for an image of type OCI
+ * It then unpacks the layers, searches for the configs folder within
+ * the unpacked content and returns it
+ */
+func GetConfigDirFromOCICatalog(ctx context.Context, ref string) (string, error) {
+
+	ociImgSrc, path, err := getOCIImgSrcFromPath(ctx, ref)
+	if err != nil {
+		return "", err
+	}
+	defer ociImgSrc.Close()
+
+	manifest, err := getManifest(ctx, ociImgSrc)
+	if err != nil {
+		return "", err
+	}
+
+	untarredLayer := ""
+	for _, layer := range manifest.LayerInfos() {
+		if !layer.EmptyLayer {
+			tmpDir := "/tmp/oc-mirror/"
+			err := os.MkdirAll(tmpDir, 0755)
+			if err != nil {
+				return "", err
+			}
+			layerSha := layer.Digest.String()
+			layerDirName := layerSha[7:]
```

Comment:
is `layerSha` length static and never less than 7?

Reply:
it is to remove the "sha256:" that stands at the beginning of the sha. the ":" will make the dir creation fail

Reply:
is it possible that `layerSha` is empty? because when it is, this code will panic